src/commands/cmd_dispatcher.c

```
@@ -224,22 +213,24 @@ int CommandDispatch
 	bool shouldCreate = should_command_create_graph (cmd) ;
-	GraphContext *gc = GraphContext_Retrieve (ctx, graph_name, true,
-			shouldCreate) ;
+	GraphContext *gc = NULL ;
+	GraphRetrieveStatus status =
+		GraphContext_Retrieve (ctx, graph_name, true, shouldCreate, false, &gc) ;
 
 	// if GraphContext is null, key access failed and an error been emitted
-	if (!gc) return REDISMODULE_ERR;
+	if (status == GraphRetrieve_FAILED) {
+		return REDISMODULE_ERR ;
+	}
```

Cppcheck (2.20.0)

[error] Common realloc mistake

(memleakOnRealloc)